### DISCUSSION TITLE
Extend Open Id connect UI for keycloak v19

### DISCUSSION THREAD

**Discussion by kluangguy1** [keycloak/keycloak#13934] [Q&A] [answered]
Dear all,
I am upgrading from keycloak v18 to v19 and notice the new field I define in the UI for my implementation of Open ID connect SPI is no longer showing. Understand by default keycloak v19 is using the new admin console with React framework.

Is there a way to extend the UI (E.G by introducing new field under OpenID connect setting section) by defining in my own implementation of the SPI JAR? Previously, I can do such customization by  adding the .html under  theme-resources\resources\partials\ and such change is only applicable when I deployed the SPI jar and for that customized openid connect SPI only.

Thanks

**Comment by ssilvert** [comment, marked as answer]
@kluangguy1 I think we already fixed that one.  Can you check the latest nightly build?
https://github.com/keycloak/keycloak/releases/tag/nightly

If it's still broken, please add an issue at https://github.com/keycloak/keycloak-ui/issues

**Reply by kluangguy1** [reply]
thanks, Am I right to say I have to make change on the typescript code in keycloak-ui project and deploy the "org.keycloak.keycloak-admin-ui-19.0.1.jar" into Keycloak v19?

**Reply by ssilvert** [reply]
You can do it that way or you could deploy your own version as a different theme.
